run_on_azureml.py

```diff
@@ -35,36 +34,31 @@ def find_or_create_compute_target(
     return target
 
 
-def parse_args(args=None):
-    parser = ArgumentParser()
-    parser.add_argument("--workspace_config", type=str, default="config.json")
-    parser.add_argument("--experiment_name", type=str, default="azureml-demo-exp")
-    parser.add_argument("--compute_target_name", type=str, default="my-compute")
-    parser.add_argument("--requirements_file", type=str, default="requirements.txt")
-    parser.add_argument("--script_path", type=str, default="simple_example/run.py")
-    parser.add_argument("--delete_compute", action="store_true", default=False)
-    args, extras = parser.parse_known_args(args)
-    args.script_args = extras
-    return args
-
-
-def main(args):
-    ws = Workspace.from_config(args.workspace_config)
-    exp = Experiment(ws, args.experiment_name)
-    target = find_or_create_compute_target(ws, args.compute_target_name)
-    env = Environment.from_pip_requirements("my-pip-env", args.requirements_file)
+def main(
+        workspace_config: str = "config.json",
+        experiment_name: str = "azureml-demo-exp",
+        compute_target_name: str = "my-compute",
+        requirements_file: str = "requirements.txt",
+        script_path: str = "simple_example/run.py",
+        delete_compute: bool = False,
+        **kwargs,
+):
+    ws = Workspace.from_config(workspace_config)
+    exp = Experiment(ws, experiment_name)
+    target = find_or_create_compute_target(ws, compute_target_name)
+    env = Environment.from_pip_requirements("my-pip-env", requirements_file)
     run_config = ScriptRunConfig(
-        source_directory=Path(args.script_path).parent,
-        script=Path(args.script_path).name,
-        arguments=args.script_args,
+        source_directory=Path(script_path).parent,
+        script=Path(script_path).name,
+        arguments=kwargs,
```

Comment:
What are `kwargs` here? I assume its a dict. Does `ScriptRunConfig` accept dictionaries? I always thought it was supposed to be a list of strings (essentially whatever you want `sys.argv` to be in your script context).

Reply:
`kwargs` is dict